src/cache/file_cache.cc

```diff
@@ -114,34 +114,101 @@ bool FileCache::Run(ui64 clean_period) {
 using namespace string;
 
 // static
-HandledHash FileCache::Hash(HandledSource code, CommandLine command_line,
-                            Version version) {
+HandledHash
+FileCache::Hash(
+  HandledSource code,
+  CommandLine command_line,
+  Version version,
+  const std::vector<string::HandledSource>& additional_sources)
+{
+  std::vector<String> hashes;
+
+  std::transform(
+    std::begin(additional_sources),
+    std::end(additional_sources),
+    std::back_inserter(hashes),
+    [] (const string::HandledSource& hs) -> String {
+      return base::Hexify(hs.str.Hash(4));
```

Comment:
The hash structure should be unified and have a fixed length - it's better to use something like [this](https://github.com/abyss7/dist-clang/blob/fa61909dd2fd8a51d697a0ce4f315c24dd3df798/src/net/test_network_service.h#L9).